### PR TITLE
Support Teleport from console & custom server avatar

### PR DESCRIPTION
**Pull request by omg-xtao**
eg: `tp @10001 x y z 3`

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by mikuyourworld** [comment]
I think you can also add custome server's name...

**Comment by alt3ri** [comment]
> I think you can also add custome server's name...

you can change server name in config.json

**Comment by mikuyourworld** [comment]
> > I think you can also add custome server's name...
> 
> you can change server name in config.json

The name in node gameserver?
I do not mean this name.

I mean the nickanme who sent message to you when you login or you use commands.. the nickname is Server now.

**Comment by alt3ri** [comment]
you can find it in game/packet/send/

**Comment by mikuyourworld** [comment]
> you can find it in game/packet/send/

Yes, I saw that..
So I am thinking that he added costome avatar, maybe he can add custome nickname together, although if I want I can change the nickname in the source.

**Comment by omg-xtao** [comment]
> > you can find it in game/packet/send/
> 
> Yes, I saw that.. So I am thinking that he added costome avatar, maybe he can add custome nickname together, although if I want I can change the nickname in the source.

Done.

File: src/main/java/emu/grasscutter/Config.java [hunk 1]
```diff
@@ -51,6 +51,7 @@ public static class RegionInfo {
 	
 	public static class GameServerOptions {
 		public String Name = "Test";
+		public String NickName = "Server";
```
Comment (memetrollsXD) [inline src/main/java/emu/grasscutter/Config.java]: Specify more, like "serverNickname"

File: src/main/java/emu/grasscutter/command/commands/TeleportCommand.java [hunk 2]
```diff
@@ -1,67 +1,88 @@
...
     public void execute(Player sender, List<String> args) {
-        if (sender == null) {
-            CommandHandler.sendMessage(null, "Run this command in-game.");
+        int target;
+        if (args.size() < (sender == null ? 4 : 3)) {
+            CommandHandler.sendMessage(sender, sender == null ? "Usage: /tp @<player id> <x> <y> <z> [scene id]" :
+                    "Usage: /tp [@<player id>] <x> <y> <z> [scene id]");
             return;
         }
+        if (args.get(0).startsWith("@")) {
```
Comment (memetrollsXD) [inline src/main/java/emu/grasscutter/command/commands/TeleportCommand.java:23]: Stay consistent with other commands and dont use @, but just the UID

**Review by memetrollsXD — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/command/commands/TeleportCommand.java (hunk 2 above)
Comment (omg-xtao) [reply src/main/java/emu/grasscutter/command/commands/TeleportCommand.java:23]: How do I handle scene id and uid if I don't use the @ prefix?

Comment (omg-xtao) [reply src/main/java/emu/grasscutter/command/commands/TeleportCommand.java:23]: All the parameters are numbers, so when you enter four parameters, how do you determine whether the extra parameters are uid or scene id?

Comment (memetrollsXD) [reply src/main/java/emu/grasscutter/command/commands/TeleportCommand.java:23]: The order of the parameters: I recommend letting xyz go first, then scene id and then UID. You can check getPlayerByUID

Comment (omg-xtao) [reply src/main/java/emu/grasscutter/command/commands/TeleportCommand.java:23]: But scene id should be optional, just like UID.

Comment (omg-xtao) [reply src/main/java/emu/grasscutter/command/commands/TeleportCommand.java:23]: Using player id should not turn the scene id into a must parameter.

**Comment by memetrollsXD** [comment]
That’s also how it goes with the /give command. Some arguments are optional
but you must provide the other argument even if they’re all optional

**Comment by omg-xtao** [comment]
> That’s also how it goes with the /give command. Some arguments are optional but you must provide the other argument even if they’re all optional

I looked at the way the give command is implemented and the code does not work when the uid and the item id are duplicated.

For example, `give 223 1`, if the uid `233` exists, it will give me an item `223` , but what I actually want is to give `233` an item 1

**Review by memetrollsXD — APPROVED** [review]
I guess there is no other way

**Comment by ghost** [comment]
Please fix your conflicts so we can merge.

**Comment by omg-xtao** [comment]
> Please fix your conflicts so we can merge.

Done.

**Pull request merged**